classes/04class/exercises/c04-iac02/danielvca/vpc.tf

```diff
@@ -0,0 +1,127 @@
+resource "aws_vpc" "devopsacademyiac" {
+  cidr_block = var.vpc_cidr
+
+  tags = {
+    Name = "devopsacademy-iac"
+  }
+}
+
+resource "aws_network_acl" "main" {
+  vpc_id = aws_vpc.devopsacademyiac.id
+
+  egress {
+    protocol   = "all"
+    rule_no    = 100
+    action     = "allow"
+    cidr_block = "10.0.0.0/16"
+    from_port  = 0
+    to_port    = 0
+  }
+
+  egress {
+    protocol   = "all"
+    rule_no    = 200
+    action     = "allow"
+    cidr_block = "180.150.58.136/32"
+    from_port  = 0
+    to_port    = 0
+  }
+
+  ingress {
+    protocol   = "all"
+    rule_no    = 300
+    action     = "allow"
+    cidr_block = "10.0.0.0/16"
+    from_port  = 0
+    to_port    = 0
+  }
+
+  ingress {
+    protocol   = "all"
+    rule_no    = 400
+    action     = "allow"
+    cidr_block = "180.150.58.136/32"
+    from_port  = 0
+    to_port    = 0
+  }
+
+}
+
+resource "aws_internet_gateway" "internet_gw" {
+  vpc_id = aws_vpc.devopsacademyiac.id
+}
+
+resource "aws_subnet" "private_subnet_a" {
+  vpc_id            = aws_vpc.devopsacademyiac.id
+  cidr_block        = var.subnets.private_subnet_a.cidr
+  availability_zone = var.subnets.private_subnet_a.az
+}
+
+resource "aws_subnet" "private_subnet_b" {
+  vpc_id            = aws_vpc.devopsacademyiac.id
+  cidr_block        = var.subnets.private_subnet_b.cidr
+  availability_zone = var.subnets.private_subnet_b.az
+}
+
+resource "aws_subnet" "public_subnet_a" {
+  vpc_id            = aws_vpc.devopsacademyiac.id
```

Comment:
For public subnets, we usually set the flag `map_public_ip_on_launch = true` so the public ip will be mapped to ec2 instances.